internal/vici/monitorSA.go

```diff
@@ -0,0 +1,278 @@
+	UniqueId               string   `json:"uniqueid"`
+}
+
+type EventIkeUpDown struct {
+	Up  bool
+	Ike map[string]*EventIkeSAUpDown
+}
+
+type EventIkeRekey struct {
+	Ike map[string]*EventIkeRekeyPair
+}
+
+type EventChildRekey struct {
+	Ike map[string]*EventIkeRekeySA
+}
+
+type EventChildUpDown struct {
+	Up  bool
+	Ike map[string]*EventIkeSAUpDown
+}
+
+type EventIkeSa struct {
+	IkeSa
+	TasksActive []string `json:"tasks-active"`
+}
+
+type EventInfo struct {
+	Up  bool
+	Ike map[string]*EventIkeSa
+}
+
+type MonitorCallBack func(event string, info interface{})
+
+func handleIkeUpDown(eventName string, callback MonitorCallBack, response map[string]interface{}) error {
+	event := &EventIkeUpDown{}
+	event.Ike = map[string]*EventIkeSAUpDown{}
+	//we need to marshall all ikes manual because json uses connections names as key
+	for name := range response {
+		value := response[name]
+		if name == "up" {
+			event.Up = true
+		} else {
+			sa := &EventIkeSAUpDown{}
+			err := convertFromGeneral(value, sa)
+			if err != nil {
+				return fmt.Errorf("convert from general: %w", err)
+			}
+			event.Ike[name] = sa
+		}
+	}
+	callback(eventName, event)
+	return nil
+}
+
+func handleIkeRekey(eventName string, callback MonitorCallBack, response map[string]interface{}) error {
+	event := &EventIkeRekey{}
+	event.Ike = map[string]*EventIkeRekeyPair{}
+	//we need to marshall all ikes manual because json uses connections names as key
+	for name := range response {
+		value := response[name]
+		sa := &EventIkeRekeyPair{}
+		err := convertFromGeneral(value, sa)
+		if err != nil {
+			return fmt.Errorf("convert from general: %w", err)
+		}
+		event.Ike[name] = sa
+	}
+	callback(eventName, event)
+	return nil
+}
+
+func handleChildUpDown(eventName string, callback MonitorCallBack, response map[string]interface{}) error {
+	event := &EventChildUpDown{}
+	event.Ike = map[string]*EventIkeSAUpDown{}
+	//we need to marshall all ikes manual because json uses connections names as key
+	for name := range response {
+		value := response[name]
+		if name == "up" {
+			event.Up = true
+		} else {
+			sa := &EventIkeSAUpDown{}
+			err := convertFromGeneral(value, sa)
+			if err != nil {
+				return fmt.Errorf("convert from general: %w", err)
+			}
+			event.Ike[name] = sa
+		}
+	}
+	callback(eventName, event)
+	return nil
+}
+
+func handleChildRekey(eventName string, callback MonitorCallBack, response map[string]interface{}) error {
+	event := &EventChildRekey{}
+	event.Ike = map[string]*EventIkeRekeySA{}
+	//we need to marshall all ikes manual because json uses connections names as key
+	for name := range response {
+		value := response[name]
+		sa := &EventIkeRekeySA{}
+		err := convertFromGeneral(value, sa)
+		if err != nil {
+			return fmt.Errorf("convert from general: %w", err)
+		}
+		event.Ike[name] = sa
+	}
+	callback(eventName, event)
+	return nil
+}
+
+func (c *ClientConn) MonitorSA(callback MonitorCallBack, watchdog time.Duration) (err error) {
```

Comment:
This looks great in terms of tracking up-down changes between our daemon checks. I'd like to get this change in sooner that later and then make a separate PR using this for event better metrics.